fabric-keybindings-v0/src/main/java/net/fabricmc/fabric/api/client/keybinding/FabricKeyBinding.java

```diff
@@ -27,23 +27,63 @@
  * {@link KeyBindingRegistry#register(FabricKeyBinding)}!
  */
 public class FabricKeyBinding extends KeyBinding {
-	protected FabricKeyBinding(Identifier id, InputUtil.Type type, int code, String category) {
-		super("key." + id.toString().replace(':', '.'), type, code, category);
+	protected FabricKeyBinding(String name, InputUtil.Type type, int code, String category) {
+		super(name, type, code, category);
 	}
 
 	public static class Builder {
-		protected final FabricKeyBinding binding;
 
-		protected Builder(FabricKeyBinding binding) {
-			this.binding = binding;
+		private InputUtil.Type type = InputUtil.Type.KEYSYM;
+
+		private String keyName = "key.fabric.unnamed";
+
+		private int code;
+
+		private String category = KeyCategory.MISC;
+
+		private Builder() {
+
+		}
+
+		public Builder name(String keyName) {
+			this.keyName = keyName;
+			return this;
+		}
+
+		public Builder name(Identifier keyName) {
+			return name("key." + keyName.toString().replace(':', '.').replace('/', '/'));
```

Comment:
Replacing / with /
what is this

Reply:
oops my chrome lagged that this is submitted after the pr got closed wut

Reply:
That was a mistake. It's supposed to change them to `.`

The idea with that is to allow longer identifiers `key.somemod.submenu.misc` which would be turned into `somemod:submenu/misc` by the Identifier.

Reply:
`.` is perfectly valid in an identifier.

Reply:
^ The modder should be able choose to use . or / as they wish

Reply:
If you're taking an identifier, and going so far as to validate and replace `:` with `.` I argue you should go the rest of the way and guard against _every_ weird input quirk that can come with it.

Reply:
```^ The modder should be able choose to use . or / as they wish```
A modder should be able to choose to use Identifier or not as they wish. Your point?

Reply:
Replacing : with . is just to change the middle character separating the namespace and path

Reply:
Well I am not against allowing the dev to use a string instead of an identifier

Reply:
Well, the PR's been close, so I guess someone is...